packages/smooth_ui_library/lib/widgets/smooth_expandable_card.dart

```diff
@@ -1,59 +1,101 @@
+
+  @override
+  void initState() {
+    super.initState();
+    _controller =
+        AnimationController(vsync: this, duration: _ANIMATION_DURATION);
+    animation = Tween<double>(begin: 0, end: pi).animate(_controller);
+  }
+
+  @override
+  void dispose() {
+    _controller.dispose();
+    super.dispose();
+  }
 
   @override
   Widget build(BuildContext context) {
-    return ChangeNotifierProvider<SingleBooleanModel>(
-      create: (BuildContext context) => SingleBooleanModel(),
-      child: Consumer<SingleBooleanModel>(
-        builder: (BuildContext context, SingleBooleanModel singleBooleanModel,
-            Widget child) {
-          return AnimatedCrossFade(
-            duration: const Duration(milliseconds: 160),
-            firstCurve: Curves.easeInOutBack,
-            secondCurve: Curves.easeInOutBack,
-            firstChild: _buildExpandedWidget(
-                singleBooleanModel, Theme.of(context), true),
-            secondChild: _buildExpandedWidget(
-                singleBooleanModel, Theme.of(context), false),
-            crossFadeState: singleBooleanModel.isActive
-                ? CrossFadeState.showSecond
-                : CrossFadeState.showFirst,
-          );
-        },
-      ),
+    return AnimatedCrossFade(
+      duration: _ANIMATION_DURATION,
+      crossFadeState: CrossFadeState.showFirst,
+      firstChild: _buildCard(),
+      secondChild: _buildCard(),
     );
   }
 
-  Widget _buildExpandedWidget(
-    final SingleBooleanModel singleBooleanModel,
-    final ThemeData themeData,
-    final bool collapsed,
-  ) {
+  Widget _buildCard() {
     return GestureDetector(
-      onTap: () => collapsed
-          ? singleBooleanModel.setActive()
-          : singleBooleanModel.setInactive(),
-      child: SmoothCard(
-        collapsed: collapsed,
-        content: content,
-        header: collapsed == true ? collapsedHeader : expandedHeader,
+      onTap: () {
+        setState(() {
+          collapsed = !collapsed;
+          animation.value == 0 ? _controller.forward() : _controller.reverse();
+        });
+      },
+      child: Padding(
+        padding: const EdgeInsets.only(
+            right: 8.0, left: 8.0, top: 4.0, bottom: 20.0),
+        child: Material(
+          elevation: 8.0,
+          shadowColor: Colors.black45,
+          borderRadius: const BorderRadius.all(Radius.circular(10.0)),
+          color: widget.background ?? Theme.of(context).colorScheme.surface,
+          child: Container(
+            padding: const EdgeInsets.all(12.0),
+            child: Column(
+              children: <Widget>[
+                Row(
+                  mainAxisSize: MainAxisSize.max,
+                  mainAxisAlignment: MainAxisAlignment.spaceBetween,
+                  crossAxisAlignment: CrossAxisAlignment.start,
+                  children: <Widget>[
+                    Container(
+                        child: collapsed
+                            ? widget.collapsedHeader
```

Comment:
Just curious: could we just use one header that works for the collapse and expanded states?

Reply:
@monsieurtanuki the expanded header is now only optional, great idea